src/app/core/services/rundown-timing.service.spec.ts

```diff
@@ -0,0 +1,428 @@
+              expectedEndEpochTime: Date.now() + rundownExpectedDurationInMs,
+              expectedDurationInMs: rundownExpectedDurationInMs,
+            }
+            const rundown: Rundown = testEntityFactory.createRundown({ timing, segments, isActive: true })
+            const testee: RundownTimingService = createTestee()
+            const expectedResult: number = 0
+
+            jasmine.clock().tick(playedDurationInMsForOnAirPart)
+
+            const result: number = testee.getRundownScheduleOffsetInMs(rundown)
+
+            expect(result).toBe(expectedResult)
+          })
+          describe('when next segment is the on air segment', () => {
+            it('returns the difference using remaining time in on air/next segment and the rest of the rundown', () => {
+              const rundownExpectedDurationInMs: number = 30_000
+              const numberOfSegments: number = 3
+              const partsPerSegment: number = 2
+              const expectedPartDurationInMs: number = rundownExpectedDurationInMs / (numberOfSegments * partsPerSegment)
+              const testEntityFactory: TestEntityFactory = new TestEntityFactory()
+              const playedDurationInMsForOnAirPart: number = expectedPartDurationInMs / 2
+              const onAirPart: Part = testEntityFactory.createPart({
+                isOnAir: true,
+                expectedDuration: expectedPartDurationInMs,
+                executedAt: Date.now(),
+              })
+              const onAirAndnextSegment: Segment = testEntityFactory.createSegment({
+                isOnAir: true,
+                isNext: true,
+                parts: [onAirPart],
+                budgetDuration: expectedPartDurationInMs * partsPerSegment,
+              })
+              const untimedSegment: Segment = testEntityFactory.createSegment({ budgetDuration: 10_000, isUntimed: true })
+              const segments: Segment[] = [
+                onAirAndnextSegment,
+                ...[...Array(numberOfSegments - 1)].map(() => testEntityFactory.createSegment({ budgetDuration: expectedPartDurationInMs * partsPerSegment })),
+                untimedSegment,
+              ]
+              const timing: BackwardRundownTiming = {
+                type: RundownTimingType.BACKWARD,
+                expectedEndEpochTime: Date.now() + rundownExpectedDurationInMs,
+                expectedDurationInMs: rundownExpectedDurationInMs,
+              }
+              const rundown: Rundown = testEntityFactory.createRundown({ timing, segments, isActive: true })
+              const testee: RundownTimingService = createTestee()
+              const expectedResult: number = 0
+
+              jasmine.clock().tick(playedDurationInMsForOnAirPart)
+
+              const result: number = testee.getRundownScheduleOffsetInMs(rundown)
+
+              expect(result).toBe(expectedResult)
+            })
+          })
+          describe('when next segment is after the on air segment', () => {
```

Comment:
A new line after a test would be greatly appreciated. There's quite a few places in this file where there is no new line after the test.

Reply:
Newlines inserted! My bad.